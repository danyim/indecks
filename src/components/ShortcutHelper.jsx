import React from 'react'
import styles from '../styles/components/ShortcutHelper.styl'

const propTypes = {}
const defaultProps = {}

const ShortcutHelper = () =>
  <section className={`${styles['shortcut-helper']}`}>
    <h2 className={`${styles.header}`}>Keyboard Shortcuts</h2>
    <div className={`${styles['shortcut-helper-content']}`}>
      <h4>Global</h4>
      <p>
        <kbd>I</kbd> - Add/Import Deck
      </p>
      <p>
        <kbd>S</kbd> - Deck Quick Selector
      </p>
      <p>
        <kbd>,</kbd> - Settings
      </p>
      <p>
        <kbd>/</kbd> - Keyboard Shortcuts
      </p>

      <h4>While viewing a deck</h4>
      <p>
        <kbd>H</kbd> - Toggle shuffle
      </p>
      <p>
        <kbd>&larr;</kbd> - Previous card
      </p>
      <p>
        <kbd>&rarr;</kbd> - Next card
      </p>
      <p>
        <kbd>SPACE</kbd> / <kbd>&uarr;</kbd> / <kbd>&darr;</kbd> - Flip card
      </p>
      <p>
        <kbd>A</kbd> - Add a card to the current deck
      </p>
      <p>
        <kbd>E</kbd> - Edit card contents
      </p>
      <p>
        <kbd>D</kbd> - Return to the deck
      </p>
    </div>
  </section>

ShortcutHelper.propTypes = propTypes
ShortcutHelper.defaultProps = defaultProps

export default ShortcutHelper
