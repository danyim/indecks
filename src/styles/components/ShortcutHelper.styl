@import '../_variables';

h2.header
  margin-top: 0
  padding-bottom: .3rem
  border-bottom: .6rem solid black
  margin-bottom: 1.0rem

h4
  margin-top: 1.0rem
  margin-bottom: .5rem

p
  margin-top: .5rem
  margin-bottom: .5rem

.settings-view
  max-width: pageWidth

.settings
  // width: 100%

.settings-content
  //
