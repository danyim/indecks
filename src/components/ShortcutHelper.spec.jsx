import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ShortcutHelper from './ShortcutHelper';

const defaultProps = {};

function setup(props = defaultProps) {
  const wrapper = shallow(<ShortcutHelper {...props} />)

  return {
    props,
    wrapper
  }
}

describe('ShortcutHelper', () => {
  it('should render self and subcomponents', () => {
    const { wrapper } = setup();
    expect(wrapper.find('section').hasClass('shortcut-helper')).toBe(true);
    expect(wrapper.find('h2').hasClass('header')).toBe(true);
    expect(wrapper.find('div').hasClass('shortcut-helper-content')).toBe(true);

    const tree = renderer.create(
      <ShortcutHelper />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
