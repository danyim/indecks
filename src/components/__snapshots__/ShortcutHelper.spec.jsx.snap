// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShortcutHelper should render self and subcomponents 1`] = `
<section
  className="shortcut-helper"
>
  <h2
    className="header"
  >
    Keyboard Shortcuts
  </h2>
  <div
    className="shortcut-helper-content"
  >
    <h4>
      Global
    </h4>
    <p>
      <kbd>
        I
      </kbd>
       - Add/Import Deck
    </p>
    <p>
      <kbd>
        S
      </kbd>
       - Deck Quick Selector
    </p>
    <p>
      <kbd>
        ,
      </kbd>
       - Settings
    </p>
    <p>
      <kbd>
        /
      </kbd>
       - Keyboard Shortcuts
    </p>
    <h4>
      While viewing a deck
    </h4>
    <p>
      <kbd>
        H
      </kbd>
       - Toggle shuffle
    </p>
    <p>
      <kbd>
        ←
      </kbd>
       - Previous card
    </p>
    <p>
      <kbd>
        →
      </kbd>
       - Next card
    </p>
    <p>
      <kbd>
        SPACE
      </kbd>
       / 
      <kbd>
        ↑
      </kbd>
       / 
      <kbd>
        ↓
      </kbd>
       - Flip card
    </p>
    <p>
      <kbd>
        A
      </kbd>
       - Add a card to the current deck
    </p>
    <p>
      <kbd>
        E
      </kbd>
       - Edit card contents
    </p>
    <p>
      <kbd>
        D
      </kbd>
       - Return to the deck
    </p>
  </div>
</section>
`
